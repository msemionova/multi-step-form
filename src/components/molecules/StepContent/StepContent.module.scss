.content {
  display: flex;
  flex-direction: column;
  flex: 1 1 100%;
  padding: 4rem 8.4rem 2rem 10rem;
  gap: 3.4rem;

  &--centered {
    justify-content: center;
  }

  i {
    align-self: center;
    width: 8rem;
    height: 8rem;
    background: no-repeat center/100% url("~/public/assets/images/icon-thanks.svg");
  }
}

.buttons {
  display: flex;
  justify-content: space-between;
}
