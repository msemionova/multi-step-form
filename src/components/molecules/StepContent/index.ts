export { default as StepContent } from './StepContent';
export type { StepContentProps } from './StepContent.types';
