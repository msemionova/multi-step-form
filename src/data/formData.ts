import { FormDataProps } from './formData.types';

const formData: FormDataProps = [
  {
    name: 'Your info',
    title: 'Personal info',
    subtitle: 'Please provide your name, email address, and phone number.',
    fieldGroups: [
      {
        type: 'inputs',
        fields: [
          {
            id: 'username',
            label: 'Name',
            placeholder: 'e.g. John Doe',
            minlength: 2,
            maxlength: 50,
            required: true
          },
          {
            id: 'email',
            label: 'Email Address',
            type: 'email',
            placeholder: 'e.g. johndoe@email.com',
            required: true
          },
          {
            id: 'phone',
            label: 'Phone Number',
            type: 'tel',
            placeholder: 'e.g. +37312345678',
            pattern: '[+]373[6-7][0-9][0-9]{6}',
            required: true
          }
        ]
      }
    ],
    buttons: {
      backBtnText: '',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Select plan',
    title: 'Select your plan',
    subtitle: 'You have the option of monthly or yearly billing.',
    fieldGroups: [
      {
        type: 'plans',
        fields: [
          {
            iconType: 'arcade',
            name: 'Arcade',
            priceMonthly: 9,
            priceYearly: 90,
            currency: 'USD'
          },
          {
            iconType: 'advanced',
            name: 'Advanced',
            priceMonthly: 12,
            priceYearly: 120,
            currency: 'USD'
          },
          {
            iconType: 'pro',
            name: 'Pro',
            priceMonthly: 15,
            priceYearly: 150,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Add-ons',
    title: 'Pick add-ons',
    subtitle: 'Add-ons help enhance your gaming experience.',
    fieldGroups: [
      {
        type: 'addons',
        fields: [
          {
            title: 'Online service',
            description: 'Access to multiplayer games',
            priceMonthly: 1,
            priceYearly: 10,
            currency: 'USD'
          },
          {
            title: 'Larger storage',
            description: 'Extra 1TB of cloud save',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          },
          {
            title: 'Customizable profile',
            description: 'Custom theme on your profile',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Summary',
    title: 'Finishing up',
    subtitle: 'Double-check everything looks OK before confirming.',
    type: 'summary',
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Confirm'
    }
  },
  {
    name: 'Thank you!',
    title: 'Thank you!',
    subtitle: `Thanks for confirming your subscription! We hope you have fun using our platform.
      If you ever need support, please feel free to email us at support@loremgaming.com.`,
    type: 'thanks'
  }
];

export default formData;
