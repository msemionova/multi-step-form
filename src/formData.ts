const formData = [
  {
    name: 'Your info',
    title: 'Personal info',
    subtitle: 'Please provide your name, email address, and phone number.',
    fieldGroups: [
      {
        type: 'inputs',
        fields: [
          {
            type: 'text',
            label: 'Name',
            placeholder: 'John Doe'
          },
          {
            type: 'email',
            label: 'Email Address',
            placeholder: 'johndoe@email.com'
          },
          {
            type: 'tel',
            label: 'Phone Number',
            placeholder: 'e.g. +1 234 567 890'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: '',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Select plan',
    title: 'Select your plan',
    subtitle: 'You have the option of monthly or yearly billing.',
    fieldGroups: [
      {
        type: 'plans',
        fields: [
          {
            iconType: 'arcade',
            name: 'Arcade',
            priceMonthly: 9,
            priceYearly: 90,
            currency: 'USD'
          },
          {
            iconType: 'advanced',
            name: 'Advanced',
            priceMonthly: 12,
            priceYearly: 120,
            currency: 'USD'
          },
          {
            iconType: 'pro',
            name: 'Pro',
            priceMonthly: 15,
            priceYearly: 150,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Add-ons',
    title: 'Pick add-ons',
    subtitle: 'Add-ons help enhance your gaming experience.',
    fieldGroups: [
      {
        type: 'addons',
        fields: [
          {
            title: 'Online service',
            description: 'Access to multiplayer games',
            priceMonthly: 1,
            priceYearly: 10,
            currency: 'USD'
          },
          {
            title: 'Larger storage',
            description: 'Extra 1TB of cloud save',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          },
          {
            title: 'Customizable profile',
            description: 'Custom theme on your profile',
            priceMonthly: 2,
            priceYearly: 20,
            currency: 'USD'
          }
        ]
      }
    ],
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Next Step'
    }
  },
  {
    name: 'Summary',
    title: 'Finishing up',
    subtitle: 'Double-check everything looks OK before confirming.',
    type: 'summary',
    buttons: {
      backBtnText: 'Go Back',
      nextBtnText: 'Confirm'
    }
  },
  {
    name: 'Thank you!',
    title: 'Thank you!',
    subtitle: `Thanks for confirming your subscription! We hope you have fun using our platform.
      If you ever need support, please feel free to email us at support@loremgaming.com.`,
    type: 'thanks'
  }
];

export default formData;
