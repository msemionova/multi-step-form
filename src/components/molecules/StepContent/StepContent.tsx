import styles from './StepContent.module.scss';
import { StepContentProps } from './StepContent.types';
import { Heading } from 'src/components/atoms';
import { FieldGroup } from 'src/components/molecules';

export default function StepContent({ stepData }: StepContentProps) {
  return (
    <div className={styles.content}>
      <Heading
        key={Math.random()}
        title={stepData.title}
        subtitle={stepData.subtitle}
      />
      {
        stepData.fieldGroups?.map(fieldGroup =>
          <FieldGroup
            key={`fieldgroup_${Math.random()}`}
            fieldGroupData={fieldGroup}
          />
        )
      }
    </div>
  );
}
