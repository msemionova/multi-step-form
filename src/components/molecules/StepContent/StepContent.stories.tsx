import type { Meta, StoryObj } from '@storybook/react';

import StepContent from './StepContent';
import formData from 'src/data/formData';

const meta: Meta<typeof StepContent> = {
  component: StepContent
};

export default meta;
type Story = StoryObj<typeof StepContent>;

export const Default: Story = {
  args: {
    stepData: formData[0]
  }
};
