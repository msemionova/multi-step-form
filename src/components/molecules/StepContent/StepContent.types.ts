import { FormStepProps } from 'src/data/formData.types';

export type StepContentProps = {
  stepData: FormStepProps
};
