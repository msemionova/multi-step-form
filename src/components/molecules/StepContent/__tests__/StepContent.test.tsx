import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import formData from 'src/formData';

import StepContent from '../StepContent';

describe('<StepContent />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StepContent data={formData}></StepContent>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
