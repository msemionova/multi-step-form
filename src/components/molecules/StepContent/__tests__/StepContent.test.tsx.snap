// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepContent /> renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="heading"
  >
    <h1>
      Personal info
    </h1>
    <p>
      Please provide your name, email address, and phone number.
    </p>
  </div>
  <div
    className="field-group"
  >
    <div
      className="input-field"
    >
      <label
        className="label"
        htmlFor="username"
      >
        Name
      </label>
      <input
        className="input"
        id="username"
        name="username"
        placeholder="e.g. John Doe"
        required={true}
        title=""
        type="text"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        className="label"
        htmlFor="email"
      >
        Email Address
      </label>
      <input
        className="input"
        id="email"
        name="email"
        placeholder="e.g. johndoe@email.com"
        required={true}
        title=""
        type="email"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        className="label"
        htmlFor="phone"
      >
        Phone Number
      </label>
      <input
        className="input"
        id="phone"
        name="phone"
        pattern="[+]373[6-7][0-9][0-9]{6}"
        placeholder="e.g. +37312345678"
        required={true}
        title=""
        type="tel"
      />
    </div>
  </div>
</div>
`;
